app/awslib.py

```diff
@@ -49,74 +42,186 @@ def _active_balancer(dns_name, region):
+    ec2_client = boto3.client('ec2', region_name=region)
+    for lb in v1_lbs:
         if lb_name in lb['LoadBalancerName'].lower():
-            #lb_name = str(lb['LoadBalancerName'])
+            lb_found = True
             print ("Processing instances for ELB %s" % lb['LoadBalancerName'])
             instances = [inst['InstanceId'] for inst in lb['Instances']]
             print ("Instances discovered: %s" % str(instances))
-
             if instances:
-                ec2 = boto3.client('ec2', region_name=region)
-                reservations = ec2.describe_instances(InstanceIds=instances)['Reservations']
-                for r in reservations:
-                    for instance in r['Instances']:
-                        if 'PublicIpAddress' in instance:
-                            instance_ips.append(instance['PublicIpAddress'])
-                        else:
-                            print ("The instance %s has no public IP" % str(instance['InstanceId']))
+                instance_ips.extend(_get_instances_public_ip(ec2_client, instances))
+    # Only look at v2 load balancers if we haven't already found the load balancer above
+    if not lb_found:
+        print("Didn't find the load balancer in the list of classic load balancers")
+        print ("Connecting to ec2 elb v2...")
+        elbv2 = boto3.client('elbv2', region_name=region)
+        print ("Connected!")
+        print ("Retrieving V2 load balancers...")
+        v2_lbs = _get_v2_lbs(elbv2, next_marker=None)
+        for lb in v2_lbs:
+            if lb_name in lb['LoadBalancerName'].lower():
+                print ("Processing target groups for ELB %s" % lb['LoadBalancerName'])
+                lb_arn = lb['LoadBalancerArn']
+                # Get the target groups for this load balancer
+                response = elbv2.describe_target_groups(LoadBalancerArn=lb_arn)
+                if 'TargetGroups' in response:
+                    lb_tgs = response['TargetGroups']
+                    # print('LB Target Groups: %s' % str(lb_tgs))
+                    for tg in lb_tgs:
+                        # print("target group: %s" % str(tg))
+                        tg_arn = tg['TargetGroupArn']
+                        target_type = tg['TargetType']
+                        instances = _get_instances_for_target_group(elbv2, tg_arn, target_type, region)
+                        if instances:
+                            instance_ips.extend(_get_instances_public_ip(ec2_client, instances))
     return instance_ips
 
-def _get_file(bucket_name, s3_path, local_path):
+
+# Get a file from S3
+def get_file(bucket_name, s3_path, local_path):
     if os.path.isfile(local_path):
         print ("Deleting current file...")
         os.remove(local_path)
         print ("Done")
     print ("Retrieving config file...")
     s3 = boto3.resource('s3')
-    s3.meta.client.download_file(bucket_name, s3_path, local_path)
+    s3.Bucket(bucket_name).download_file(s3_path, local_path)
```

Comment:
How do we know that this succeeded?

Reply:
if the file exists? :)

Reply:
Yeah looks like the some of the instances where this method is called do actually check when they try to open the file.  Some do not.